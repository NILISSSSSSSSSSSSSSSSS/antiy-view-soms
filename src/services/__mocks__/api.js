// GET 根据ID获取资产详情

export function getAssetHardWareById (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: { asset: { stringId: '-------------' } }
  })
}

export function batchRelation (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: { }
  })
}

export function claimTaskBatch (params) {
  return Promise.resolve({
    head: { code: '200' }
  })
}

export function querySuddenList (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {
      items: [
        {
          antiyVulnId: '12345'
        }
      ],
      totalRecords: 1
    }
  })
}
export function getList (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {
      items: [{}],
      totalRecords: 1
    }
  })
}
export function scanUrl (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {
      onProgress: 10,
      errorMessage: '',
      findAssetNums: 10
    }
  })
}
export function getBugUsers (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: []
  })
}
export function onBackSubmit (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: []
  })
}
export function getAdvise (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {}
  })
}
export function getVulnCpe (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {
      items: [],
      totalRecords: 1
    }
  })
}
export default {
  scanUrl,
  getAdvise,
  onBackSubmit,
  getBugUsers,
  getList,
  getAssetHardWareById,
  batchRelation,
  claimTaskBatch,
  querySuddenList,
  getVulnCpe
}
