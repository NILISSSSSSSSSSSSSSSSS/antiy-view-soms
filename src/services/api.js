import asset from './assets'
import * as workOrderApi from './workOrder'
import * as systemApi from './system'
import * as safeApi from './safe'
import * as organizationApi from './organization'
import * as taskPlan from './taskPlan'
import * as taskPlanReport from './taskPlanReport'
import * as bug from './bug'
import log from  './log'
import * as knowledge from  './knowledge'
import * as inspection from  './inspection'
import alarm from './alarm'
import * as baseSetting from  './baseSetting'
import * as indexPage from  './indexPage'
import * as workBench from  './workBench'
import * as reportForm from  './reportForm'
import  personalCenter from  './personalCenter'
import patch from './patch'
import * as defend from  './defend'
import * as examine from  './examine'
export default {
  ...patch,
  ...workBench,
  ...indexPage,
  ...asset,
  ...workOrderApi,
  ...systemApi,
  ...taskPlan,
  ...taskPlanReport,
  ...organizationApi,
  ...bug,
  ...log,
  ...knowledge,
  ...alarm,
  ...baseSetting,
  ...safeApi,
  ...inspection,
  ...reportForm,
  ...personalCenter,
  ...defend,
  ...examine
}
