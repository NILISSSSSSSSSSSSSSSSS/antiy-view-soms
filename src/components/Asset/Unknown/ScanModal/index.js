import React, { Component } from 'react'
import { CommonModal } from '@c'
import api from '@/services/api'
import { debounce } from 'lodash'

const formLayout = {
  labelCol: {
    span: 8
  },
  wrapperCol: {
    span: 14
  }
}
const pageSize = -1 // 代表查所有数据
export default class ScanModal extends Component {
  constructor (props) {
    super(props)
    this.state = {
      // 网段
      networkSegment: [],
      // 端口
      portGroup: []
    }
  }

  componentDidMount () {
    this.getNetworkSegmentList()
    this.getPortGroupList()
  }
  // 搜索网段
  onSearchNetworkSegment = (value) => {
    this.onSearchNetworkSegment({ item: value })
  }
  // 搜索端口
  onSearchPortGroup = (value) => {
    console.log(value)
    this.getPortGroupList({ information: value })
  }
  // 获取网段列表
  getNetworkSegmentList = (filter) => {
    // 上一次的请求未返回时，不进行请求
    if(this.networkSegmentLoading){
      return
    }
    this.networkSegmentLoading = true
    api.getNetsegment({ ...filter, pageSize }).then(res=>{
      const { items = [] } = res.body || {}
      this.setState({ networkSegment: items })
      this.networkSegmentLoading = false
    }).catch(()=>{
      this.networkSegmentLoading = false
    })
  }
  // 获取端口列表
  getPortGroupList = (filter) => {
    // 上一次的请求未返回时，不进行请求
    if(this.portGroupLoading){
      return
    }
    this.portGroupLoading = true
    api.getParamPortList({ ...filter, pageSize }).then((res)=>{
      const { items = [] } = res.body || {}
      this.setState({ portGroup: items })
      this.portGroupLoading = false
    }).catch(()=>{
      this.portGroupLoading = false
    })
  }
  onSubmit = debounce((values) => {
    const { onSubmit } = this.props
    onSubmit && onSubmit(values)
  })

  render () {
    const { visible, onClose } = this.props
    const { networkSegment, portGroup } = this.state
    const fields = [
      {
        type: 'select',
        key: 'netSegmentId',
        name: '扫描网段',
        showSearch: true,
        mode: 'multiple',
        config: { name: 'portGroupName', value: 'primaryKey' },
        rules: [
          { required: true, message: '请选择扫描网段' }
        ],
        placeholder: '请选择扫描网段',
        data: networkSegment
      },
      {
        type: 'select',
        key: 'portGroupId',
        name: '扫描端口组',
        showSearch: true,
        rules: [
          { required: true, message: '请选择扫描端口组' }
        ],
        config: { name: 'portGroupName', value: 'primaryKey' },
        placeholder: '请选择扫描端口组',
        data: portGroup
      },

      {
        type: 'select',
        key: 'alarm',
        name: '探测后是否产生告警',
        defaultValue: 2,
        placeholder: '请选择',
        data: [
          { label: '是', value: 1 },
          { label: '否', value: 2 }
        ]
      }
    ]
    return (
      <CommonModal
        title='资产探测'
        type="form"
        visible={ visible }
        formLayout={ formLayout }
        fields={ fields }
        oktext='确定'
        value={ this.onSubmit }
        onClose={ onClose }
      />
    )
  }
}
