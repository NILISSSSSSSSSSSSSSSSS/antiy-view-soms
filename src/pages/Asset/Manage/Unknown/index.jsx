import React, { Component } from 'react'
import { TableBtns, Table } from '@c/index'
import { withRouter } from 'dva/router'
import Tooltip from '@/components/common/CustomTooltip'
import ScanModal from '@/components/Asset/Unknown/ScanModal'
import { transliteration, analysisUrl, cacheSearchParameter, evalSearchParam } from '@/utils/common'
import moment from 'moment'
import { WAIT_REGISTER, ASSET_SOURCE } from '@a/js/enume'
import Search from '@/components/common/Search'
import Scan from '@/components/common/Scan'
import ExportModal from '@/components/common/ExportModal'
import { assetsPermission } from '@a/permission'
import api from '@/services/api'
import hasAuth from '@/utils/auth'
import { message } from 'antd'
const pageSize = 10
export class Unknown extends Component{
  constructor (props){
    super(props)
    // ?status=2&assetSource=1&work=true&conditionShow=true
    const param = analysisUrl(props.location.search)
    this.expandedStatus = param.conditionShow === 'true'
    this.id = param.status === '2' ? param.id : null
    this.enterControl = param.work === 'true' ? true : void 0
    this.state = {
      filter: {
        // assetSource: param.assetSource ? parseInt(param.assetSource, 10) : void 0,
        sortOrder: {}
      },
      exportVisible: false, // 导出弹窗状态
      scanVisible: false, // 扫描进度弹窗
      modalData: { // 资产探测探测
        visible: false
      },
      selectedRowKeys: [],
      scanParams: {}, // 调教探测的参数
      currentPage: 1,
      pageSize,
      scanData: {}, // 实时探测信息
      total: 0,
      list: []
    }
  }
  onSearch = (values) => {
    const { pageSize, filter } = this.state
    this.setState({ filter: { ...filter, currentPage: 1, ...values } })
    this.getList({ ...filter, ...values }, 1, pageSize)
  }
  /**
   * 重置查询
   */
  onReset = () => {
    this.setState({
      filter: {
        // assetSource: void 0,
        sortOrder: {}
      }
    }, ()=>this.onSearch({}))
  }
  /**
   * 获取资产列表
   * @param filter
   * @param currentPage
   * @param pageSize
   * @param isSave {Boolean} 是否保存数据
   * @return {*}
   */
  getList = (filter, currentPage, pageSize, isSave = true) => {
    const { firstEnter = [], multipleQuery, sortOrder, assetSource, enterControl } = filter || {}
    const [ firstEnterStartTime, firstEnterEndTime ] = firstEnter
    const { columnKey: sortName, order  } = sortOrder
    const pagingParameter = {
      currentPage,
      pageSize
    }
    const params = {
      multipleQuery
      // assetSource: this.state.filter.assetSource || assetSource || ''
    }
    const paramsOyher = {
      sortName,
      assetStatusList: [ WAIT_REGISTER.value ], // 永远只能查询待登记状态的资产
      unknownAssets: true,
      firstEnterStartTime: firstEnterStartTime ? firstEnterStartTime.valueOf() : null,
      firstEnterEndTime: firstEnterEndTime ? firstEnterEndTime.valueOf() : null,
      sortOrder: order === 'ascend' ? 'ASC' : 'DESC',
      enterControl: this.enterControl || enterControl
    }
    // 保存最新查询条件，供导出时使用
    this.exportParams = { ...params, ...pagingParameter, ...paramsOyher, assetId: this.id }
    cacheSearchParameter([{
      page: pagingParameter,
      parameter: {
        ...params,
        firstEnterStartTime,
        firstEnterEndTime,
        sortOrder: sortOrder || {},
        enterControl
      }
    }], this.props.history, 1, '2')
    return api.getAssetList(this.exportParams).then(res=>{
      const { items: list, totalRecords: total } = res.body || {}
      isSave && this.setState({ list, total, currentPage })
      return res
    })
  }
  onChange = (pagination, filters, sorter) => {
    const { current: currentPage, pageSize } = pagination
    const { filter } = this.state
    const _filter = { ...filter, sortOrder: sorter }
    this.setState({ currentPage, pageSize, filter: _filter })
    this.getList(_filter, currentPage, pageSize)
  }
  // 单个不予登记
  notRegisterSingle = (record) => {
    this.notRegisterSubmit([record.stringId])
  }
  // 批量不予登记
  notRegisterBatch = () => {
    const { selectedRowKeys } = this.state
    this.notRegisterSubmit(selectedRowKeys)
  }
  /**
   * 不予登记提交
   * @param assetId
   */
  notRegisterSubmit = (assetIds) => {
    let assetList = []
    assetIds.forEach(item=>{
      const content = {
        taskId: null,
        assetId: item
      }
      assetList.push(content)
    })
    api.assetNoRegister(assetList).then(res => {
      if(res && res.head && res.head.code === '200' ){
        const { filter, currentPage, pageSize } = this.state
        message.success('操作成功！')
        this.setState({ selectedRowKeys: [] })
        this.getList(filter, currentPage, pageSize)
      }
    }).catch(() => {})
  }
  onSubmitScan = (values) => {
    const scanParams = { ...values }
    if(values.alarm === 2){
      scanParams.alarm = false
    }else {
      scanParams.alarm = true
    }
    this.onSubmitAssetDetect(scanParams)
  }
  /**
   * 提交探测
   * @param params
   */
  onSubmitAssetDetect = (params) => {
    api.submitAssetDetect(params).then(()=>{
      this.setState({ modalData: { visible: false }, scanVisible: true, scanParams: params })
    })
  }
  /**
   * 资产探测前提
   * */
  // assetDetectPrecondition = () => {
  //   api.getAssetDetect().then((res)=>{
  //     if(res.body){
  //       this.setState({ scanVisible: true, scanData: res.body || {} })
  //     }else {
  //       this.setState({ modalData: { visible: true }, scanVisible: false })
  //     }
  //   })
  // }
  /**
   * 导出事件，导出之前先去查询一下最新数据
   **/
  exportExcel = () => {
    const { filter } = this.state
    this.getList(filter, void 0, void 0, false).then(res=>{
      const { totalRecords: total } = res.body || {}
      if(total){
        this.setState({ exportVisible: true })
      }else {
        message.error('暂无数据可导出!')
      }
    })
  }
  /**
   * 操作之前，验证资产状态是否被改变
   * @param record {Object} 资产信息
   * @param operation {Number} 操作类型
   *    1 变更
   *    2 登记
   *    3 不予登记
   *    4 拟退役
   *  */
  verificationAssetStatus = (operation, record) => {
    return  api.verificationAssetStatus({ assetId: record.stringId, operation }).catch((err)=>{
      this.onReset()
      return Promise.reject(err)
    })
  }
  render () {
    const { currentPage, pageSize, total, list, filter: { assetSource }, selectedRowKeys, modalData: { visible }, scanVisible, exportVisible, scanParams } = this.state
    //漏洞扫描
    // const scanConfig = {
    //   isAssets: true,
    //   equipments: 0,
    //   interval: 5000,
    //   params: scanParams,
    //   scanUrl: 'getAssetDetect',
    //   title: '资产探测扫描',
    //   visible: scanVisible,
    //   onClose: () => {
    //     this.setState({ scanVisible: false })
    //   },
    //   onBackstage: () => {
    //     this.setState({ scanVisible: false })
    //   },
    //   onStopFind: () => {
    //     api.stopAssetDetect().then(()=>{
    //       this.setState({ scanVisible: false })
    //     })
    //   },
    //   goTo: () => {
    //     api.stopAssetDetect()
    //     this.onReset()
    //     this.searchResetKey = Math.random()
    //     this.setState({ scanVisible: false })
    //   },
    //   // 探测数据变化了，进行回调，重置列表数据
    //   dataChangeCallback: () => {
    //     this.searchResetKey = Math.random()
    //     this.onReset()
    //   }
    // }
    const columns = [
      {
        title: 'IP',
        dataIndex: 'ips',
        render: text=><Tooltip title={text}>{text}</Tooltip>
      },
      {
        title: 'MAC',
        dataIndex: 'macs',
        render: text=><Tooltip title={text}>{text}</Tooltip>
      },
      {
        title: '资产来源',
        dataIndex: 'assetSourceName',
        render: (text, record)=> <Tooltip title={text}>{text}</Tooltip>
      },
      {
        title: '首次发现时间',
        dataIndex: 'gmtCreate',
        render: (text) => {
          return ( text ? <span className="tabTimeCss">{moment(text).format('YYYY-MM-DD HH:mm:ss')}</span> : '--')
        },
        sorter: true
      }
    ]
    const actions = [
      {
        type: 'link',
        label: '查看',
        // check: hasAuth(assetsPermission.ASSET_INFO_VIEW),
        to: (record)=>`/asset/manage/detail?id=${transliteration(record.stringId)}`
      },
      {
        type: 'btn',
        label: '不予登记',
        confirm: { onOk: this.notRegisterSingle, text: '是否将该资产列为"不予登记"状态?' },
        check: (record)=>record.assetStatus === WAIT_REGISTER.value && hasAuth(assetsPermission.ASSET_BYDJ)
      },
      {
        type: 'btn',
        label: '登记',
        // to: (record)=>`/asset/manage/register?source=appear&id=${transliteration(record.stringId)}&categoryModel=${record.categoryModel}&register=again`,
        onClick: (record)=>{
          const { push } = this.props.history
          this.verificationAssetStatus(2, record).then(()=>{
            push(`/asset/manage/register?source=appear&id=${transliteration(record.stringId)}&categoryModel=${record.categoryModel}&register=again`)
          })
        },
        check: (record)=>record.assetStatus === WAIT_REGISTER.value && hasAuth(assetsPermission.ASSET_DJ)
      }
    ]
    const leftBtns = [
      { label: '导出', check: hasAuth(assetsPermission.ASSET_EXPORT), onClick: this.exportExcel }
      // { label: '资产探测', check: hasAuth(assetsPermission.ASSET_UNKNOWN_FIND), onClick: this.assetDetectPrecondition }
    ]
    const rightBtns = [{ label: '不予登记', check: hasAuth(assetsPermission.ASSET_BYDJ), checkUsable: ()=> selectedRowKeys.length ? null : '请选择资产！', confirm: { onOk: this.notRegisterBatch, text: '是否将该资产列为"不予登记"状态?' }  }]
    const defaultFields = [ { label: '综合查询', maxLength: 30, key: 'multipleQuery', span: 12, type: 'input', placeholder: '请输入IP/MAC' }]
    const fieldList = [
      // { label: '资产来源', key: 'assetSource', initialValue: assetSource, type: 'select', data: ASSET_SOURCE },
      { label: '首次发现时间', key: 'firstEnter', type: 'dateRange' }
    ]
    const rowSelection = {
      selectedRowKeys,
      onChange: (selectedRowKeys)=>{
        this.setState({ selectedRowKeys })
      }
    }
    return(
      <div className="unknown-containar">
        <div className="search-bar">
          <Search key={this.searchResetKey}
            defaultFields={defaultFields}
            fieldList={fieldList}
            onReset={this.onReset}
            onSubmit={this.onSearch}
            setDefaultExpandedStatus={this.expandedStatus}
            wrappedComponentRef={search => { search && (this.searchForm = search.props.form) }}
          />
        </div>
        <div className='table-wrap'>
          <TableBtns leftBtns={leftBtns} rightBtns={rightBtns}/>
          <Table rowSelection={rowSelection} columns={columns} page={{ currentPage, pageSize, total }} dataSource={list} actions={actions} onChange={this.onChange}/>
        </div>
        { visible && <ScanModal visible={visible} onSubmit={this.onSubmitScan} onClose={()=>this.setState({ modalData: { visible: false } })}/> }
        {/* { scanVisible && <Scan  {...scanConfig}/>} */}
        {/* 导出 */}
        {
          exportVisible && <ExportModal
            exportModal={{
              exportVisible, //弹框显示
              searchValues: this.exportParams, //搜索条件
              total, //数据总数
              threshold: 5000, //阈值
              url: '/api/v1/asset/export/file' //下载地址
            }}
            handleCancelExport={() => this.setState({ exportVisible: false })} />
        }
      </div>
    )
  }
  componentDidMount () {
    const { filter, currentPage, pageSize } = this.state
    const { id } = analysisUrl(this.props.location.search) // id为外部链接跳转过来时，直接查询某一条资产

    const { list } = evalSearchParam(this, {}, false) || {}
    //判断是否存有数据
    if (list && list[1]) {
      const { page: { pageSize, currentPage }, parameter } = list[1]
      this.setState({
        pageSize,
        currentPage
      }, () => {
        this.searchForm.setFieldsValue( parameter )
        this.getList({ ...parameter, stringId: id }, currentPage, pageSize)
      })
    } else {
      this.getList({ ...filter, stringId: id }, currentPage, pageSize)
    }
  }
}
export default withRouter(Unknown)
