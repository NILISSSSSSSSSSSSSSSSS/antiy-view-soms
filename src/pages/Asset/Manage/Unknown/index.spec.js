import React from 'react'
import { shallow } from 'enzyme'
import Unknown from './index'
import api from '@/services/api'
import { BrowserRouter }    from 'react-router-dom'
import PropTypes from 'prop-types'
// jest.mock('@/services/api')
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {
      search: '?id=asPTSbUoE0fZbilfBaL15w%3D%3D'
    },
    match: {}
  }
}

const createContext = {
  context: { router },
  childContextTypes: { router: PropTypes.object.isRequired }
}

const head = { code: '200' }
describe('未知资产列表 <Unknown />', () => {
  let wrapper
  beforeEach(()=>{
    // wrapper = shallow(<Router><Unknown { ...props }/></Router>)
    wrapper = shallow(<Unknown />, createContext).dive().dive()
  })
  it('未知资产列表的正常渲染', () => {
    expect(wrapper.find('.unknown-containar')).toHaveLength(1)
  })
  it('onSearch ==>查询条件的查询事件', () => {
    const instance = wrapper.instance()
    const spyFunction_onSearch = jest.spyOn(instance, 'onSearch')
    spyFunction_onSearch({ name: '查询' })
    expect(spyFunction_onSearch).toHaveBeenCalled()
    const state = wrapper.state()
    expect(state.filter.name).toEqual('查询')
  })
  it('getList ==>获取资产列表', () => {
    const instance = wrapper.instance()
    const spyFunction_getList = jest.spyOn(instance, 'getList')
    api.getAssetList = jest.fn()
    const spyFunction_getAssetList = jest.spyOn(api, 'getAssetList')
    spyFunction_getAssetList.mockReturnValue(Promise.resolve({
      head,
      body: {
        items: [],
        totalRecords: 0
      }
    }))
    // 调用getList
    const result = spyFunction_getList({ name: '查询', sortOrder: { columnKey: 'key', order: 'ascend' } }, 1, 10)
    result.then(()=>{})
    expect(spyFunction_getList).toHaveBeenCalled()
    expect(spyFunction_getAssetList).toHaveBeenCalled()
  })
  it('onChange ==> 表格翻页事件', ()=>{
    const spyFunction_onChange = jest.spyOn(wrapper.instance(), 'onChange')
    const pagination = { current: 2, pageSize: 10 }
    spyFunction_onChange(pagination, {}, { columnKey: 'keyName', order: 'ascend' })
    expect(spyFunction_onChange).toHaveBeenCalled()
    const state = wrapper.state()
    expect(state.currentPage).toEqual(2)
    expect(state.filter.sortOrder).toEqual({ columnKey: 'keyName', order: 'ascend' })
  })
  it('notRegisterSingle ==> 单个不予登记', ()=>{
    const spyFunction_notRegisterSingle = jest.spyOn(wrapper.instance(), 'notRegisterSingle')
    const spyFunction_notRegisterSubmit = jest.spyOn(wrapper.instance(), 'notRegisterSubmit')
    spyFunction_notRegisterSingle({})
    expect(spyFunction_notRegisterSubmit).toHaveBeenCalled()
  })
  it('notRegisterBatch ==> 批量不予登记', ()=>{
    const spyFunction_notRegisterBatch = jest.spyOn(wrapper.instance(), 'notRegisterBatch')
    const spyFunction_notRegisterSubmit = jest.spyOn(wrapper.instance(), 'notRegisterSubmit')
    spyFunction_notRegisterBatch([])
    expect(spyFunction_notRegisterSubmit).toHaveBeenCalled()
  })
  it('notRegisterSubmit ==> 不予登记提交', ()=>{
    const spyFunction_notRegisterSubmit = jest.spyOn(wrapper.instance(), 'notRegisterSubmit')
    api.assetNoRegister = jest.fn(c=>c=>c)
    api.assetNoRegister.mockReturnValue(Promise.resolve({
      head
    }))
    wrapper.setState({
      selectedRowKeys: ['2']
    })
    spyFunction_notRegisterSubmit('1')
    expect(wrapper.state().selectedRowKeys).toEqual(['2'])

    api.assetNoRegister('1').then((res)=>{
      if(res.head.code === '200'){
        wrapper.setState({
          selectedRowKeys: []
        })
        expect(wrapper.state().selectedRowKeys).toEqual([])
      }
    })
  })
  it('onSubmitScan ==> 探测弹窗提交事件', ()=>{
    const spyFunction_onSubmitScan = jest.spyOn(wrapper.instance(), 'onSubmitScan')
    spyFunction_onSubmitScan({ alarm: 1 })
    spyFunction_onSubmitScan({ alarm: 2 })
  })
  it('onSubmitAssetDetect ==> 提交弹窗请求', ()=>{
    const spyFunction_onSubmitAssetDetect = jest.spyOn(wrapper.instance(), 'onSubmitAssetDetect')
    api.submitAssetDetect = jest.fn()
    api.submitAssetDetect.mockReturnValue(Promise.resolve({}))
    spyFunction_onSubmitAssetDetect({ alarm: true })
    expect(spyFunction_onSubmitAssetDetect).toHaveBeenCalled()
  })
  it('assetDetectPrecondition ==> 资产探测前提', ()=>{
    const spyFunction_assetDetectPrecondition = jest.spyOn(wrapper.instance(), 'assetDetectPrecondition')
    api.getAssetDetect = jest.fn()
    api.getAssetDetect.mockReturnValue(Promise.resolve({ head, body: null }))
    spyFunction_assetDetectPrecondition({ alarm: true })
    api.getAssetDetect.mockReturnValue(Promise.resolve({ head, body: {} }))
    spyFunction_assetDetectPrecondition({ alarm: true })
  })
  it('exportExcel ==> 导出事件', ()=>{
    const spyFunction_exportExcel = jest.spyOn(wrapper.instance(), 'exportExcel')
    const spyFunction_getList = jest.spyOn(wrapper.instance(), 'getList')
    spyFunction_getList.mockReturnValue(Promise.resolve({ head, body: { totalRecords: 0 } }))
    spyFunction_exportExcel()
    spyFunction_getList.mockReturnValue(Promise.resolve({ head, body: { totalRecords: 1 } }))
    spyFunction_exportExcel()
  })
  it('快照', ()=>{
    const app = wrapper.debug()
    expect(app).toMatchSnapshot()
  })
})
