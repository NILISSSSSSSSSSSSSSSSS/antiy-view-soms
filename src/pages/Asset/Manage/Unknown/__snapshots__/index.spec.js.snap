// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`未知资产列表 <Unknown /> 快照 1`] = `
"<div className=\\"unknown-containar\\">
  <div className=\\"search-bar\\">
    <Form(Search2) defaultFields={{...}} fieldList={{...}} onSubmit={[Function]} setDefaultExpandedStatus={false} />
  </div>
  <div className=\\"table-wrap\\">
    <TableBtns leftBtns={{...}} rightBtns={{...}} />
    <CustomTable rowSelection={{...}} columns={{...}} page={{...}} dataSource={{...}} actions={{...}} onChange={[Function]} />
  </div>
</div>"
`;
